nntrainer/compiler/compiler.h

```diff
@@ -97,12 +113,124 @@ class GraphInterpreter {
   virtual GraphRepresentation deserialize(std::istream &in) = 0;
 };
 
-class GraphInterpreter;
+/**
+ * @brief Pure Virtual class for ExecutableGraph
+ *
+ * @note this class is an interface to executable graph as actual graph would be
+ * different framework by framework
+ *
+ */
+class ExecutableGraph {
+  /// NYI!
+};
 
 /**
- * @brief a simple scaffoliding function
+ * @brief Graph Representation
+ *
+ * @note this class must have the graph representation of the most expanded
+ * version as far as nntrainer supports.
+ *
+ * GraphRepresentation contains the operation configurations, connections and
+ * buffer(this might not...)
  *
  */
-void hello_world();
+class GraphRepresentation {
+public:
+  /**
+   * @brief Construct a new Graph Representation object
+   *
+   * @param interpreter_ interpreter to use
+   * @param compiler_ compiler to use
+   */
+  GraphRepresentation(std::shared_ptr<GraphInterpreter> interpreter_ = nullptr,
+                      std::shared_ptr<GraphCompiler> compiler_ = nullptr) :
```

Comment:
I thought GraphCompiler use graphRepresentation. This looks the opposite now.

Reply:
Yes, this is initial design commit, only the last commit is valid for this PR. I'll rebase now :)